lib/arel/sql_to_arel/pg_query_visitor.rb

```diff
@@ -288,7 +288,7 @@ def visit_ColumnRef(fields:)
           column_reference = visit(column_reference, :operator)
           table[column_reference]
         else
-          Arel::Nodes::UnboundColumnReference.new visited_fields.join('.')
+          Arel::Nodes::UnqualifiedColumn.new(Arel::Attribute.new(nil, visited_fields.join('.')))
```

Comment:
Could you add a `case when` for 1, 2 and else? The else branch should raise an exception because "posts.id.title" is never a thing